src/services/webhookServices.ts

```
@@ -1,2 +1,59 @@
-//placeholder for webhook services
-//to trigger CI pipeline
+import axios from "axios";
+import logger from "../utils/logger";
+import {
+  WebhookPayload,
+  ServiceConfig,
+  DispatchPayload,
+} from "../models/webhookModels";
+import { EVENT_TYPE } from "../utils/constants";
+
+export const createWebhookService = (services: ServiceConfig[]) => {
+  const dispatchToService = async (
+    service: ServiceConfig,
+    payload: DispatchPayload,
+  ): Promise<void> => {
+    try {
+      await axios.post(service.url, payload);
+      logger.info(`Successfully dispatched to ${service.name}`);
+    } catch (error) {
+      logger.error(`Error dispatching to ${service.name}:`, error);
+      throw error;
```

If we throw the error here then it will bubble up to the map function it's being called from [here](https://github.com/Extheoisah/hxckr-webhook-handler/blob/2894f0e5fdf29981494c77684444e4cb93fdb3e0/src/services/webhookServices.ts#L42) which means that if one service call fails then none of the other services will receive the call.

---

Yes is that not supposed to be how we should should propagate the error, since all service connected are using the same data. 

I am thinking I'm terms of if the data failes to send to one, others should also fail. 

What do you suggest in this scenario?

---

Lets catch up on our call this afternoon 👍